core/azure-core/src/main/java/com/azure/core/implementation/configuration/Configuration.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    /**
+     * Gets the value of the configuration.
+     *
+     * @param name Name of the configuration.
+     * @return Value of the configuration if found, otherwise null.
+     */
+    public String get(String name) {
+        return getOrLoad(name);
+    }
+
+    /**
+     * Gets the value of the configuration converted to {@code T}.
+     *
+     * If no configuration is found the default is returned.
+     *
+     * @param name Name of the configuration.
+     * @param defaultValue Value to return if the configuration isn't found.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise the default value is returned.
+     */
+    public <T> T get(String name, T defaultValue) {
+        return convertOrDefault(getOrLoad(name), defaultValue);
+    }
+
+    /**
+     * Gets the converted value of the configuration.
+     *
+     * @param name Name of the configuration.
+     * @param converter Converter used to map the configuration to {@code T}.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise null.
+     */
+    public <T> T get(String name, Function<String, T> converter) {
+        String value = getOrLoad(name);
+        if (ImplUtils.isNullOrEmpty(value)) {
+            return null;
+        }
+
+        return converter.apply(value);
+    }
+
+    /**
+     * Attempts to get the value of the configuration from the configuration store, if the value isn't found then it
+     * attempts to load it from the runtime parameters then the environment variables.
+     *
+     * If no configuration is found null is returned.
+     *
+     * @param name Name of the configuration.
+     * @return The configuration value from either the configuration store, runtime parameters, or environment
+     * variable, in that order, if found, otherwise null.
+     */
+    private String getOrLoad(String name) {
+        if (configurations.containsKey(name)) {
+            return configurations.get(name);
+        }
+
+        return load(name);
+    }
+
+    /**
+     * Attempts to load the configuration from the environment.
+     *
+     * The runtime parameters are checked first followed by the environment variables. If the configuration is found
+     * the value is loaded into the configuration store and if a configuration with the same name already exists this
+     * will update it to the loaded value.
+     *
+     * @param name Name of the configuration.
+     * @return If found the loaded configuration, otherwise null.
+     */
+    private String load(String name) {
+        if (loadFrom(name, System::getProperty)) {
+            logger.asInformational().log(LOADED_FROM_RUNTIME, name);
+        } else if (loadFrom(name, System::getenv)) {
+            logger.asInformational().log(LOADED_FROM_ENVIRONMENT, name);
+        }
```

Comment:
To make the logging more useful, you should also include the value that was loaded in the output.